packages/story-editor/src/components/library/panes/shared/hooks/useNestedRovingTabIndex.js

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from '@googleforcreators/react';
+import { useKeyDownEffect } from '@googleforcreators/design-system';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../app';
+
+const SELECTOR = '[tabIndex="-1"]:not(:disabled)';
+
+// @todo Combine this with `useRovingTabIndex` so that if depth is set, this file is used, otherwise the default one is used.
+// @todo Move common helpers into a shared file between the two.
```

Comment:
@barklund @merapi WDYT of this approach for the nested buttons roving tab index? 

This is an adjustment of `useNestedRovingTabIndex` so a lot of similar stuff but with variations.

This still needs some polishing but should work with local Media library. I'm also thinking of combining `useRovingTabIndex` and `useNestedRovingTabIndex` so that in the code `useRovingTabIndex` could always be used but if the `depth` is set, it would just use the nested version in the background.
Some assumptions it has:
- The current button (or any focusable element) has `tabIndex` `0`,  all the others have `-1` (kind of as it works currently)
- The selector for looking into nested elements to shift focus includes the `tabIndex=-1` meaning that the first element with `tabIndex=-1` will be selected. This means that multiple nested elements with `tabIndex=-1` wouldn't potentially work as expected. Unlikely use case anyway though.
- If nesting level changes but the `depth` is not updated, the code breaks (true now as well -- currently it works only with focusable siblings -- it has sometimes broken because those focusable elements have been wrapped into another element)

Thoughts?